source/code/source/playfab/PlayFabWinHttpPlugin.cpp

```diff
@@ -372,10 +370,15 @@ namespace PlayFab
     void PlayFabWinHttpPlugin::SetPredefinedHeaders(const CallRequestContainer& requestContainer, HINTERNET hRequest)
     {
         UNREFERENCED_PARAMETER(requestContainer);
-        WinHttpAddRequestHeaders(hRequest, L"Accept: application/json", -1, 0);
-        WinHttpAddRequestHeaders(hRequest, L"Content-Type: application/json; charset=utf-8", -1, 0);
-        WinHttpAddRequestHeaders(hRequest, (L"X-PlayFabSDK: " + std::wstring(PlayFabSettings::versionString.begin(), PlayFabSettings::versionString.end())).c_str(), -1, 0);
-        WinHttpAddRequestHeaders(hRequest, L"X-ReportErrorAsSuccess: true", -1, 0);
+        headerNoErrors = !TryAddHeader(hRequest, L"Accept: application/json", -1, 0) &&
+            !TryAddHeader(hRequest, L"Content-Type: application/json; charset=utf-8", -1, 0) &&
+            !TryAddHeader(hRequest, (L"X-PlayFabSDK: " + std::wstring(PlayFabSettings::versionString.begin(), PlayFabSettings::versionString.end())).c_str(), -1, 0) &&
+            !TryAddHeader(hRequest, L"X-ReportErrorAsSuccess: true", -1, 0);
+    }
+
+    bool PlayFabWinHttpPlugin::TryAddHeader(HINTERNET hRequest, LPCWSTR lpszHeaders, DWORD dwHeadersLength, DWORD dwModifiers)
```

Comment:
it looks like all callers of TryAddHeader pass -1 for `dwHeadersLength` and 0 for `dwModifiers`

you could just hardcode those values inside of the TryAddHeader helper until someone actually needs to pass something different. One of the advantages of wrapping external APIs with helper methods is being able to simplify the parameter lists they take (which makes the code calling that wrapper easier to read)